storage/remote/write_handler.go

```diff
@@ -92,6 +104,45 @@ func NewWriteHandler(logger *slog.Logger, reg prometheus.Registerer, appendable
 	return h
 }
 
+// validateV1LabelValueLength checks label values in v1 protobuf format before ToLabels() is called.
+// This prevents panics during deserialization of oversized labels.
+func validateV1LabelValueLength(labels []prompb.Label) (labelName string, valueLength int, exceeded bool) {
+	for _, l := range labels {
+		if len(l.Value) > maxLabelValueLength {
+			return l.Name, len(l.Value), true
+		}
+	}
+	return "", 0, false
+}
+
+// validateV2LabelValueLength checks label values in v2 protobuf format before ToLabels() is called.
+// This prevents panics during deserialization of oversized labels.
+func validateV2LabelValueLength(labelsRefs []uint32, symbols []string) (labelName string, valueLength int, exceeded bool) {
+	// labelsRefs are pairs: [nameRef, valueRef, nameRef, valueRef, ...]
+	for i := 0; i < len(labelsRefs); i += 2 {
```

Comment:
It feels `ToLabel` could be extended to be limit aware. Would be simpler and cheaper, WDYT?